[GTK] Unreviewed, build fix for Ubuntu 20.04 after 262210@main

dpino opened the pull request:
#### 84f5108371981e77ec6c245eb788e2176efe69fc
<pre>
[GTK] Unreviewed, build fix for Ubuntu 20.04 after 262210@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=252988">https://bugs.webkit.org/show_bug.cgi?id=252988</a>

Variable &apos;m_displayForCompositing&apos; was not declared for condition PLATFORM(GTK) &amp;&amp; USE(EGL).

* Source/WebKit/WebProcess/WebProcess.h:

Canonical link: <a href="https://commits.webkit.org/262405@main">https://commits.webkit.org/262405@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b5a659cf6775b8219ec853c12bc2ab24bb612284

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1468 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1498 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1548 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2397 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/1299 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1559 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1558 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/1415 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1481 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/1345 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/1326 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2236 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/1351 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/1309 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/1270 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/1340 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/1349 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/2419 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/1393 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/1236 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/1310 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/1316 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1435 "Built successfully") | | | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/156 "Failed to push commit to Webkit repository") | | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/WebProcess/WebProcess.h:
```diff
@@ -706,7 +706,7 @@ class WebProcess : public AuxiliaryProcess
     WeakHashMap<WebCore::UserGestureToken, uint64_t> m_userGestureTokens;
 
-#if PLATFORM(WAYLAND)
+#if PLATFORM(WAYLAND) || (PLATFORM(GTK) && USE(EGL))
```
This is weird, WAYLAND platform is only defined by the GTK port and EGL is requirement of the wayland platform.

dpino replied:
The only bot affected by this error is `Ubuntu 20.04`. This bot builds with `no-experimental-features` and with system libraries (no `flatpak` neither `jhbuild minimal`). I think the build error might be related to that.

dpino replied:
The reason why the build is failing in the Ubuntu 20.04 is because this bot builds with the flag ```--cmakeargs=-DENABLE_WAYLAND_TARGET=OFF```. Since WAYLAND is not enabled the variable `m_displayForCompositing` is not defined. But https://github.com/webkit/webkit/compare/d2424c0fc67f%5E..d2424c0fc67f#diff-d4c35777d04d911675eb95adf2a7cc767a915c0ed07068fcd02a7318b51a0015R126 makes use of this variable if PLATFORM is GTK and EGL is used.

I think the fix is correct. If `m_displayForCompositing` can be used under the condition `PLATFORM(GTK) && USE(EGL)` the variable should also be defined under that condition and not rely only on whether WAYLAND is enabled.

dpino committed:
[GTK] Unreviewed, build fix for Ubuntu 20.04 after 262210@main
https://bugs.webkit.org/show_bug.cgi?id=252988

Variable 'm_displayForCompositing' was not declared for condition PLATFORM(GTK) && USE(EGL).

* Source/WebKit/WebProcess/WebProcess.h:

Canonical link: https://commits.webkit.org/262405@main
